### PR TITLE
Add AndroidX migration docs

### PR DESCRIPTION
**Pull request by blasten**
This doc replaces the existing page https://flutter.dev/docs/development/packages-and-plugins/androidx-compatibility.  

I added this page under development since it targets all types of Flutter projects.

### CODE REVIEW

File: src/docs/development/androidx-migration.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      alt="An animation of the bottom-up page transition on Android" />
+8. That is it! You successfully migrated your project to AndroidX. 
+
+Finally, if you migrated a plugin, publish the new AndroidX version to pub and update 
+your `CHANGELOG.md` to indicate that this new version is compatible with AndroidX.
+
+
+### How do I migrate my existing module project to AndroidX?
+
+If your project is a module and contains a `.android` directory. Then,
```
Comment (sfshaza2) [inline src/docs/development/androidx-migration.md]: => If your project is a module and contains a `.android` directory, add the following
line to `pubspec.yml`:

**Comment by sfshaza2** [comment]
Also, I see that Travis is encountering some link problems. Can you fix?
```
http://localhost:4002/docs/codelabs/layout-basics
- (1224:0) 'Text class' => http://localhost:4002/docs/codelabs/layout-basics#text-class (HTTP 200 but missing anchor)
http://localhost:4002/docs/development/androidx-migration
- (999:41) 'develope..' => http://localhost:4002/docs/development/developer.android.com/studio (HTTP 404)
- (999:41) 'develope..' => http://localhost:4002/docs/development/developer.android.com/studio (HTTP 404)
```

File: src/docs/development/androidx-migration.md (hunk 1 above)
Comment (blasten) [reply src/docs/development/androidx-migration.md]: Done. Thanks!

File: src/docs/development/androidx-migration.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+### What if I don’t migrate my app or module to AndroidX?
+
+Your app may continue to work. However, combining AndroidX and Support artifacts 
+is generally not recommended because it can result in dependency conflicts or 
+other kind of Gradle failures. As a result, as more plugins migrate to AndroidX,
+plugins depending on Android core libraries are likely to cause build failures.
+
+### What if my app is migrated to AndroidX, but not all of the plugins I use?
+
+The Flutter tool uses Jetifier to automatically migrate Flutter plugins using
```
Comment (mklim) [inline src/docs/development/androidx-migration.md]: Have we enabled building the plugins as AARs by default? My most recent understanding is that we still have compatibility problems here, but I could be missing something.

Comment (blasten) [reply src/docs/development/androidx-migration.md]: yeah, it's now enabled by default. However, plugins are built as AAR only when the build failed and the heuristic we have for detecting AndroidX failure is true.

Comment (mklim) [reply src/docs/development/androidx-migration.md]: Yeah, LGTM! :)

**Pull request merged**